Simplify loading_cache_test and use manual_clock

bhalevy opened the pull request:
This series exposes a Clock template parameter for loading_cache so that the test could use
the manual_clock rather than the lowres_clock, since relying on the latter is flaky.

In addition, the test load function is simplified to sleep some small random time and co_return the expected string,
rather than reading it from a real file, since the latter's timing might also be flaky, and it out-of-scope for this test.

Fixes #20322

* The test was flaky forever, so backport is required for all live versions.

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14102/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14102/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: boost/bloom_filter_test
:small_blue_diamond: boost/loading_cache_test
:small_blue_diamond: boost/reader_concurrency_semaphore_test
:small_blue_diamond: boost/view_build_test
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13660/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5647/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2533/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/903/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/307/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14102/)
#### Build Details:
- Duration: 4 hr 51 min
- Builder: spider5.cloudius-systems.com

denesb commented on test/raft/replication.hh:
```diff
@@ -1235,7 +1235,7 @@ future<> raft_cluster<Clock>::check_rpc_config(::check_rpc_config cc) {
         BOOST_CHECK(node.id < _servers.size());
         co_await seastar::async([&] {
-            CHECK_EVENTUALLY_EQUAL(_servers[node.id].rpc->known_peers(), as);
+            BOOST_CHECK(eventually_true([&] { return _servers[node.id].rpc->known_peers() == as; }));
```
Why not use the new function here too?

bhalevy replied:
Because BOOST_REQUIRE_EQUAL doesn't know how to print `raft::server_address_set`

denesb commented on test/lib/eventually.hh:
```diff
@@ -49,5 +51,16 @@ bool eventually_true(noncopyable_function<bool ()> f) {
+template <typename T>
+void REQUIRE_EVENTUALLY_EQUAL(std::function<T()> a, T b) {
+    eventually_true([&] { return a() == b; });
+    BOOST_REQUIRE_EQUAL(a(), b);
```
This creates an opportunity for a confusing test failure, where `eventually_true()` fails, but the `BOOST_REQUIRE_EQUAL` succeeds here. I think it has a low enough chance of happening that we can let it slip.

bhalevy replied:
The outcome is still equivalent to eventually_true succeeding (eventually)

denesb commented on test/boost/bloom_filter_test.cc:
```diff
@@ -96,9 +96,9 @@ SEASTAR_TEST_CASE(test_sstable_manager_auto_reclaim_and_reload_of_bloom_filter)
         // smallest filter in the reclaimed list, which is sst1's bloom filter.
         dispose_and_stop_tracking_bf_memory(std::move(sst3), sst_mgr);
-        REQUIRE_EVENTUALLY_EQUAL(sst1->filter_memory_size(), sst1_bf_memory);
+        REQUIRE_EVENTUALLY_EQUAL<uint64_t>([&] { return sst1->filter_memory_size(); }, sst1_bf_memory);
```
Why is explicit instantiation needed, isn't the compiler able to deduce the type?

bhalevy replied:
Unfortunately not.
```
test/boost/bloom_filter_test.cc:99:9: error: no matching function for call to 'REQUIRE_EVENTUALLY_EQUAL'
   99 |         REQUIRE_EVENTUALLY_EQUAL([&] { return sst1->filter_memory_size(); }, sst1_bf_memory);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
./test/lib/eventually.hh:77:6: note: candidate template ignored: could not match 'std::function<T ()>' against '(lambda at test/boost/bloom_filter_test.cc:99:34)'
   77 | void REQUIRE_EVENTUALLY_EQUAL(std::function<T()> a, T b, sleep_fn sleep = seastar_sleep_fn{}) {
      |      ^
1 error generated.
```

denesb commented on test/lib/eventually.hh:
```diff
@@ -15,8 +15,16 @@
+    future<> operator()(std::chrono::milliseconds ms) const {
+        return seastar::sleep(ms);
+    }
+};
```
This could be just a simple function too.

bhalevy replied:
okay.  Will also move it out of line.

denesb reviewed (approved):
LGTM, minor comments only.

bhalevy commented:
In v2 ([048175d](https://github.com/scylladb/scylladb/commit/048175d3067334eb52057607e7e117b40eade7aa)):
- rebased
- converted also `reader_concurrency_semaphore_test`
- moved sleep functions out of line as simple, global lambda-assigned std::functions.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14312/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14312/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: boost/bloom_filter_test
:small_blue_diamond: boost/loading_cache_test
:small_blue_diamond: boost/reader_concurrency_semaphore_test
:small_blue_diamond: boost/view_build_test
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1044/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2674/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13802/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5789/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/446/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14312/)
#### Build Details:
- Duration: 8 hr 2 min
- Builder: i-0e60feadaae2c9932 (m5ad.12xlarge)

bhalevy commented:
> #### 🔴 CI State: FAILURE
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14312/) ✅ - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14312/) **The following new/updated tests ran 100 times for each mode:** 🔹 boost/bloom_filter_test 🔹 boost/loading_cache_test 🔹 boost/reader_concurrency_semaphore_test 🔹 boost/view_build_test ✅ - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1044/) ✅ - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2674/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13802/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5789/) ✅ - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/446/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14312/)
> 

https://github.com/scylladb/scylladb/issues/22243

Seems like a benign issue and unrelated to this change

mykaul commented:
@scylladb/scylla-maint - please review for inclusion - it fails our CI from time to time.

bhalevy commented:
@scylladb/scylla-maint please consider merging
This fixes ci flakiness (https://github.com/scylladb/scylladb/issues/20322 that's still hitting us from time to time) and therefore should be allowed as part of the effort to unfreeze next.

bhalevy commented:
> Dequeued, breaks the build

In what way?
Can you provide a link and/or excerpt of the failure?
Why wasn't that hit in CI?

bhalevy commented:
> > Dequeued, breaks the build
> 
> In what way? Can you provide a link and/or excerpt of the failure? Why wasn't that hit in CI?

ping @avikivity 
It is unclear what problem was caused by this change
and what needs to be fixed to get it in.

avikivity commented:
> > > Dequeued, breaks the build
> > 
> > In what way? Can you provide a link and/or excerpt of the failure? Why wasn't that hit in CI?
> 
> ping @avikivity 
> It is unclear what problem was caused by this change

Compile error


> and what needs to be fixed to get it in.
> 

It needs to compile without errors

bhalevy commented:
In v3 ([df4c7c8](https://github.com/scylladb/scylladb/commit/df4c7c81ac213ed8636e43e80dfcfb77a4d7a660)):
- rebased
- moved sleep function definitions to new test/lib/eventually.cc files
  - and updated dependencies in configure.py

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:x: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14656/)
#### Build Details:
- Duration: 33 min
- Builder: spider6.cloudius-systems.com

bhalevy commented:
> #### 🔴 CI State: FAILURE
> ❌ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14656/)
> 
> #### Build Details:
> * Duration: 33 min
> * Builder: spider6.cloudius-systems.com

```
[3265/3894] Linking CXX executable Dev/scylla
FAILED: Dev/scylla 
: && /usr/lib64/ccache/clang++ -I/usr/include/p11-kit-1 -Wno-backend-plugin -fprofile-use="/jenkins/workspace/scylla-master/scylla-ci/scylla/build/profile.profdata"  -O2 -Xlinker --build-id=sha1 --ld-path=ld.lld -dynamic-linker=/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////lib64/ld-linux-x86-64.so.2 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/Dev/seastar/libseastar.so /usr/lib64/libboost_program_options.so /usr/lib64/libboost_thread.so /usr/lib64/libcares.so /usr/lib64/libfmt.so.10.2.1 -L/usr/lib64 -llz4 CMakeFiles/scylla_version.dir/Dev/release.cc.o CMakeFiles/scylla.dir/Dev/main.cc.o -o Dev/scylla  -Wl,-rpath,/jenkins/workspace/scylla-master/scylla-ci/scylla/build/Dev/seastar  Dev/libscylla-main.a  api/Dev/libapi.a  alternator/Dev/libalternator.a  db/Dev/libdb.a  cdc/Dev/libcdc.a  compaction/Dev/libcompaction.a  cql3/Dev/libcql3.a  data_dictionary/Dev/libdata_dictionary.a  gms/Dev/libgms.a  index/Dev/libindex.a  lang/Dev/liblang.a  ent/ldap/Dev/libldap.a  message/Dev/libmessage.a  mutation/Dev/libmutation.a  mutation_writer/Dev/libmutation_writer.a  raft/Dev/libraft.a  readers/Dev/libreaders.a  redis/Dev/libredis.a  repair/Dev/librepair.a  replica/Dev/libreplica.a  schema/Dev/libschema.a  service/Dev/libservice.a  sstables/Dev/libsstables.a  streaming/Dev/libstreaming.a  test/perf/Dev/libtest-perf.a  tools/Dev/libtools.a  transport/Dev/libtransport.a  types/Dev/libtypes.a  utils/Dev/libutils.a  /usr/lib64/libp11-kit.so  Dev/seastar/libseastar.so  /usr/lib64/libyaml-cpp.so  /usr/lib64/libboost_program_options.so.1.83.0  -Xlinker --push-state -Xlinker --whole-archive  ent/encryption/Dev/libscylla_encryption.a  -Xlinker --pop-state  /usr/lib64/libssl.so  /usr/lib64/libcrypto.so  /jenkins/workspace/scylla-master/scylla-ci/scylla/kmipc/kmipc-2.1.0t-rhel84_64/lib/libkmip.a  test/lib/Dev/libtest-lib.a  -Xlinker --push-state -Xlinker --whole-archive  audit/Dev/libscylla_audit.a  -Xlinker --pop-state  -Xlinker --push-state -Xlinker --whole-archive  auth/Dev/libscylla_auth.a  -Xlinker --pop-state  ent/ldap/Dev/libldap.a  /usr/lib64/libldap.so  /usr/lib64/liblber.so  /usr/lib64/libldap.so  /usr/lib64/libcrypt.so  cdc/Dev/libcdc.a  compaction/Dev/libcompaction.a  mutation_writer/Dev/libmutation_writer.a  -Xlinker --push-state -Xlinker --whole-archive  dht/Dev/libscylla_dht.a  -Xlinker --pop-state  index/Dev/libindex.a  -Xlinker --push-state -Xlinker --whole-archive  locator/Dev/libscylla_locator.a  -Xlinker --pop-state  message/Dev/libmessage.a  gms/Dev/libgms.a  sstables/Dev/libsstables.a  readers/Dev/libreaders.a  schema/Dev/libschema.a  -Xlinker --push-state -Xlinker --whole-archive  tracing/Dev/libscylla_tracing.a  -Xlinker --pop-state  Dev/libscylla-main.a  -Xlinker --push-state -Xlinker --whole-archive  Dev/libscylla-zstd.a  -Xlinker --pop-state  /usr/lib64/libzstd.so  abseil/absl/strings/Dev/libabsl_cord.a  abseil/absl/strings/Dev/libabsl_cordz_info.a  abseil/absl/strings/Dev/libabsl_cord_internal.a  abseil/absl/strings/Dev/libabsl_cordz_functions.a  abseil/absl/strings/Dev/libabsl_cordz_handle.a  abseil/absl/crc/Dev/libabsl_crc_cord_state.a  abseil/absl/crc/Dev/libabsl_crc32c.a  abseil/absl/crc/Dev/libabsl_crc_internal.a  abseil/absl/crc/Dev/libabsl_crc_cpu_detect.a  abseil/absl/strings/Dev/libabsl_str_format_internal.a  /usr/lib64/liblz4.a  /usr/lib64/libzstd.a  service/Dev/libservice.a  node_ops/Dev/libnode_ops.a  service/Dev/libservice.a  node_ops/Dev/libnode_ops.a  raft/Dev/libraft.a  repair/Dev/librepair.a  streaming/Dev/libstreaming.a  replica/Dev/libreplica.a  abseil/absl/container/Dev/libabsl_raw_hash_set.a  abseil/absl/hash/Dev/libabsl_hash.a  abseil/absl/hash/Dev/libabsl_city.a  abseil/absl/types/Dev/libabsl_bad_variant_access.a  abseil/absl/hash/Dev/libabsl_low_level_hash.a  abseil/absl/types/Dev/libabsl_bad_optional_access.a  abseil/absl/container/Dev/libabsl_hashtablez_sampler.a  abseil/absl/profiling/Dev/libabsl_exponential_biased.a  abseil/absl/synchronization/Dev/libabsl_synchronization.a  abseil/absl/debugging/Dev/libabsl_stacktrace.a  abseil/absl/synchronization/Dev/libabsl_graphcycles_internal.a  abseil/absl/synchronization/Dev/libabsl_kernel_timeout_internal.a  abseil/absl/debugging/Dev/libabsl_symbolize.a  abseil/absl/debugging/Dev/libabsl_debugging_internal.a  abseil/absl/base/Dev/libabsl_malloc_internal.a  abseil/absl/debugging/Dev/libabsl_demangle_internal.a  abseil/absl/time/Dev/libabsl_time.a  abseil/absl/strings/Dev/libabsl_strings.a  abseil/absl/strings/Dev/libabsl_strings_internal.a  abseil/absl/strings/Dev/libabsl_string_view.a  abseil/absl/base/Dev/libabsl_throw_delegate.a  abseil/absl/numeric/Dev/libabsl_int128.a  abseil/absl/base/Dev/libabsl_base.a  abseil/absl/base/Dev/libabsl_raw_logging_internal.a  abseil/absl/base/Dev/libabsl_log_severity.a  abseil/absl/base/Dev/libabsl_spinlock_wait.a  -lrt  abseil/absl/time/Dev/libabsl_civil_time.a  abseil/absl/time/Dev/libabsl_time_zone.a  -lsystemd  /usr/lib64/libz.so  /usr/lib64/libdeflate.so  types/Dev/libtypes.a  utils/Dev/libutils.a  /usr/lib64/libyaml-cpp.so  /usr/lib64/libcryptopp.so  /usr/lib64/libgnutls.so  /usr/lib64/libboost_regex.so.1.83.0  /usr/lib64/libicui18n.so  /usr/lib64/libicuuc.so  -ldl  /usr/lib64/libboost_unit_test_framework.so.1.83.0  Dev/seastar/libseastar_perf_testing.so  /usr/lib64/libjsoncpp.so.1.9.5  db/Dev/libdb.a  data_dictionary/Dev/libdata_dictionary.a  cql3/Dev/libcql3.a  transport/Dev/libtransport.a  cql3/Dev/libcql3.a  transport/Dev/libtransport.a  lang/Dev/liblang.a  /usr/lib64/liblua-5.4.so  -lm  rust/Dev/libwasmtime_bindings.a  rust/librust_combined.a  /usr/lib64/libsnappy.so.1.1.10  mutation/Dev/libmutation.a  Dev/seastar/libseastar.so  /usr/lib64/liblz4.so  /usr/lib64/libxxhash.so && :
ld.lld: error: undefined symbol: seastar_sleep_fn
>>> referenced by random_schema.cc
>>>               random_schema.cc.o:(seastar::noncopyable_function<void ()>::direct_vtable_for<seastar::futurize<std::invoke_result<tests::random_schema::create_with_cql(cql_test_env&)::$_0>::type>::type seastar::async<tests::random_schema::create_with_cql(cql_test_env&)::$_0>(seastar::thread_attributes, tests::random_schema::create_with_cql(cql_test_env&)::$_0&&)::'lambda'()>::call(seastar::noncopyable_function<void ()> const*)) in archive test/lib/Dev/libtest-lib.a
>>> referenced by random_schema.cc
>>>               random_schema.cc.o:(seastar::noncopyable_function<void ()>::direct_vtable_for<seastar::futurize<std::invoke_result<tests::random_schema::create_with_cql(cql_test_env&)::$_0>::type>::type seastar::async<tests::random_schema::create_with_cql(cql_test_env&)::$_0>(seastar::thread_attributes, tests::random_schema::create_with_cql(cql_test_env&)::$_0&&)::'lambda'()>::call(seastar::noncopyable_function<void ()> const*)) in archive test/lib/Dev/libtest-lib.a
>>> referenced by random_schema.cc
>>>               random_schema.cc.o:(seastar::noncopyable_function<void ()>::direct_vtable_for<seastar::futurize<std::invoke_result<tests::random_schema::create_with_cql(cql_test_env&)::$_0>::type>::type seastar::async<tests::random_schema::create_with_cql(cql_test_env&)::$_0>(seastar::thread_attributes, tests::random_schema::create_with_cql(cql_test_env&)::$_0&&)::'lambda'()>::call(seastar::noncopyable_function<void ()> const*)) in archive test/lib/Dev/libtest-lib.a
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```
Hmm, `ninja dev-build` passed forcme locally.
I'm not sure why ai haven't emcountered the above failure.
Maybe it needs building with cmake?

bhalevy commented:
In v4 ([32b7cab](https://github.com/scylladb/scylladb/commit/32b7cab917dabca67d8a54d3ba3055261bdf6b40)):
- added test/lib/eventually.cc to `test-lib` target_sources in test/lib/CMakeLists.txt

denesb commented:
> > #### 🔴 CI State: FAILURE
> > ❌ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14656/)
> > #### Build Details:
> > 
> > * Duration: 33 min
> > * Builder: spider6.cloudius-systems.com
> 
> ```
> [3265/3894] Linking CXX executable Dev/scylla
> FAILED: Dev/scylla 
> : && /usr/lib64/ccache/clang++ -I/usr/include/p11-kit-1 -Wno-backend-plugin -fprofile-use="/jenkins/workspace/scylla-master/scylla-ci/scylla/build/profile.profdata"  -O2 -Xlinker --build-id=sha1 --ld-path=ld.lld -dynamic-linker=/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////lib64/ld-linux-x86-64.so.2 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/Dev/seastar/libseastar.so /usr/lib64/libboost_program_options.so /usr/lib64/libboost_thread.so /usr/lib64/libcares.so /usr/lib64/libfmt.so.10.2.1 -L/usr/lib64 -llz4 CMakeFiles/scylla_version.dir/Dev/release.cc.o CMakeFiles/scylla.dir/Dev/main.cc.o -o Dev/scylla  -Wl,-rpath,/jenkins/workspace/scylla-master/scylla-ci/scylla/build/Dev/seastar  Dev/libscylla-main.a  api/Dev/libapi.a  alternator/Dev/libalternator.a  db/Dev/libdb.a  cdc/Dev/libcdc.a  compaction/Dev/libcompaction.a  cql3/Dev/libcql3.a  data_dictionary/Dev/libdata_dictionary.a  gms/Dev/libgms.a  index/Dev/libindex.a  lang/Dev/liblang.a  ent/ldap/Dev/libldap.a  message/Dev/libmessage.a  mutation/Dev/libmutation.a  mutation_writer/Dev/libmutation_writer.a  raft/Dev/libraft.a  readers/Dev/libreaders.a  redis/Dev/libredis.a  repair/Dev/librepair.a  replica/Dev/libreplica.a  schema/Dev/libschema.a  service/Dev/libservice.a  sstables/Dev/libsstables.a  streaming/Dev/libstreaming.a  test/perf/Dev/libtest-perf.a  tools/Dev/libtools.a  transport/Dev/libtransport.a  types/Dev/libtypes.a  utils/Dev/libutils.a  /usr/lib64/libp11-kit.so  Dev/seastar/libseastar.so  /usr/lib64/libyaml-cpp.so  /usr/lib64/libboost_program_options.so.1.83.0  -Xlinker --push-state -Xlinker --whole-archive  ent/encryption/Dev/libscylla_encryption.a  -Xlinker --pop-state  /usr/lib64/libssl.so  /usr/lib64/libcrypto.so  /jenkins/workspace/scylla-master/scylla-ci/scylla/kmipc/kmipc-2.1.0t-rhel84_64/lib/libkmip.a  test/lib/Dev/libtest-lib.a  -Xlinker --push-state -Xlinker --whole-archive  audit/Dev/libscylla_audit.a  -Xlinker --pop-state  -Xlinker --push-state -Xlinker --whole-archive  auth/Dev/libscylla_auth.a  -Xlinker --pop-state  ent/ldap/Dev/libldap.a  /usr/lib64/libldap.so  /usr/lib64/liblber.so  /usr/lib64/libldap.so  /usr/lib64/libcrypt.so  cdc/Dev/libcdc.a  compaction/Dev/libcompaction.a  mutation_writer/Dev/libmutation_writer.a  -Xlinker --push-state -Xlinker --whole-archive  dht/Dev/libscylla_dht.a  -Xlinker --pop-state  index/Dev/libindex.a  -Xlinker --push-state -Xlinker --whole-archive  locator/Dev/libscylla_locator.a  -Xlinker --pop-state  message/Dev/libmessage.a  gms/Dev/libgms.a  sstables/Dev/libsstables.a  readers/Dev/libreaders.a  schema/Dev/libschema.a  -Xlinker --push-state -Xlinker --whole-archive  tracing/Dev/libscylla_tracing.a  -Xlinker --pop-state  Dev/libscylla-main.a  -Xlinker --push-state -Xlinker --whole-archive  Dev/libscylla-zstd.a  -Xlinker --pop-state  /usr/lib64/libzstd.so  abseil/absl/strings/Dev/libabsl_cord.a  abseil/absl/strings/Dev/libabsl_cordz_info.a  abseil/absl/strings/Dev/libabsl_cord_internal.a  abseil/absl/strings/Dev/libabsl_cordz_functions.a  abseil/absl/strings/Dev/libabsl_cordz_handle.a  abseil/absl/crc/Dev/libabsl_crc_cord_state.a  abseil/absl/crc/Dev/libabsl_crc32c.a  abseil/absl/crc/Dev/libabsl_crc_internal.a  abseil/absl/crc/Dev/libabsl_crc_cpu_detect.a  abseil/absl/strings/Dev/libabsl_str_format_internal.a  /usr/lib64/liblz4.a  /usr/lib64/libzstd.a  service/Dev/libservice.a  node_ops/Dev/libnode_ops.a  service/Dev/libservice.a  node_ops/Dev/libnode_ops.a  raft/Dev/libraft.a  repair/Dev/librepair.a  streaming/Dev/libstreaming.a  replica/Dev/libreplica.a  abseil/absl/container/Dev/libabsl_raw_hash_set.a  abseil/absl/hash/Dev/libabsl_hash.a  abseil/absl/hash/Dev/libabsl_city.a  abseil/absl/types/Dev/libabsl_bad_variant_access.a  abseil/absl/hash/Dev/libabsl_low_level_hash.a  abseil/absl/types/Dev/libabsl_bad_optional_access.a  abseil/absl/container/Dev/libabsl_hashtablez_sampler.a  abseil/absl/profiling/Dev/libabsl_exponential_biased.a  abseil/absl/synchronization/Dev/libabsl_synchronization.a  abseil/absl/debugging/Dev/libabsl_stacktrace.a  abseil/absl/synchronization/Dev/libabsl_graphcycles_internal.a  abseil/absl/synchronization/Dev/libabsl_kernel_timeout_internal.a  abseil/absl/debugging/Dev/libabsl_symbolize.a  abseil/absl/debugging/Dev/libabsl_debugging_internal.a  abseil/absl/base/Dev/libabsl_malloc_internal.a  abseil/absl/debugging/Dev/libabsl_demangle_internal.a  abseil/absl/time/Dev/libabsl_time.a  abseil/absl/strings/Dev/libabsl_strings.a  abseil/absl/strings/Dev/libabsl_strings_internal.a  abseil/absl/strings/Dev/libabsl_string_view.a  abseil/absl/base/Dev/libabsl_throw_delegate.a  abseil/absl/numeric/Dev/libabsl_int128.a  abseil/absl/base/Dev/libabsl_base.a  abseil/absl/base/Dev/libabsl_raw_logging_internal.a  abseil/absl/base/Dev/libabsl_log_severity.a  abseil/absl/base/Dev/libabsl_spinlock_wait.a  -lrt  abseil/absl/time/Dev/libabsl_civil_time.a  abseil/absl/time/Dev/libabsl_time_zone.a  -lsystemd  /usr/lib64/libz.so  /usr/lib64/libdeflate.so  types/Dev/libtypes.a  utils/Dev/libutils.a  /usr/lib64/libyaml-cpp.so  /usr/lib64/libcryptopp.so  /usr/lib64/libgnutls.so  /usr/lib64/libboost_regex.so.1.83.0  /usr/lib64/libicui18n.so  /usr/lib64/libicuuc.so  -ldl  /usr/lib64/libboost_unit_test_framework.so.1.83.0  Dev/seastar/libseastar_perf_testing.so  /usr/lib64/libjsoncpp.so.1.9.5  db/Dev/libdb.a  data_dictionary/Dev/libdata_dictionary.a  cql3/Dev/libcql3.a  transport/Dev/libtransport.a  cql3/Dev/libcql3.a  transport/Dev/libtransport.a  lang/Dev/liblang.a  /usr/lib64/liblua-5.4.so  -lm  rust/Dev/libwasmtime_bindings.a  rust/librust_combined.a  /usr/lib64/libsnappy.so.1.1.10  mutation/Dev/libmutation.a  Dev/seastar/libseastar.so  /usr/lib64/liblz4.so  /usr/lib64/libxxhash.so && :
> ld.lld: error: undefined symbol: seastar_sleep_fn
> >>> referenced by random_schema.cc
> >>>               random_schema.cc.o:(seastar::noncopyable_function<void ()>::direct_vtable_for<seastar::futurize<std::invoke_result<tests::random_schema::create_with_cql(cql_test_env&)::$_0>::type>::type seastar::async<tests::random_schema::create_with_cql(cql_test_env&)::$_0>(seastar::thread_attributes, tests::random_schema::create_with_cql(cql_test_env&)::$_0&&)::'lambda'()>::call(seastar::noncopyable_function<void ()> const*)) in archive test/lib/Dev/libtest-lib.a
> >>> referenced by random_schema.cc
> >>>               random_schema.cc.o:(seastar::noncopyable_function<void ()>::direct_vtable_for<seastar::futurize<std::invoke_result<tests::random_schema::create_with_cql(cql_test_env&)::$_0>::type>::type seastar::async<tests::random_schema::create_with_cql(cql_test_env&)::$_0>(seastar::thread_attributes, tests::random_schema::create_with_cql(cql_test_env&)::$_0&&)::'lambda'()>::call(seastar::noncopyable_function<void ()> const*)) in archive test/lib/Dev/libtest-lib.a
> >>> referenced by random_schema.cc
> >>>               random_schema.cc.o:(seastar::noncopyable_function<void ()>::direct_vtable_for<seastar::futurize<std::invoke_result<tests::random_schema::create_with_cql(cql_test_env&)::$_0>::type>::type seastar::async<tests::random_schema::create_with_cql(cql_test_env&)::$_0>(seastar::thread_attributes, tests::random_schema::create_with_cql(cql_test_env&)::$_0&&)::'lambda'()>::call(seastar::noncopyable_function<void ()> const*)) in archive test/lib/Dev/libtest-lib.a
> clang++: error: linker command failed with exit code 1 (use -v to see invocation)
> ```
> 
> Hmm, `ninja dev-build` passed for me locally. I'm not sure why ai haven't emcountered the above failure. Maybe it needs building with cmake?

See https://github.com/scylladb/scylla-pkg/commit/8fbe808ed7e6823771424a8b68dcec33405d92fa

bhalevy commented:
I also verified build with cmake using:
```
$ ./configure.py --use-cmake --mode=dev --cflags=-g
$ cmake --build build --config Dev
```

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14682/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14682/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: boost/bloom_filter_test
:small_blue_diamond: boost/loading_cache_test
:small_blue_diamond: boost/reader_concurrency_semaphore_test
:small_blue_diamond: boost/view_build_test
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/6080/)
:x: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2957/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/14091/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14682/)
#### Failed Tests (3/38026):
- [test_rolling_upgrade](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14682/testReport/junit/enterprise_upgrade_test/TestEnterpriseRollingUpgrade/Tests___dtest_with_topology_changes___test_rolling_upgrade) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_rolling_upgrade)
- [test_rolling_upgrade](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14682/testReport/junit/enterprise_upgrade_test/TestEnterpriseRollingUpgrade/Tests___dtest_with_tablets___test_rolling_upgrade) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_rolling_upgrade)
- [test_rolling_upgrade](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14682/testReport/junit/enterprise_upgrade_test/TestEnterpriseRollingUpgrade/Tests___dtest___test_rolling_upgrade) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_rolling_upgrade)
#### Build Details:
- Duration: 11 hr
- Builder: i-0075c4dd2b520f080 (m5ad.8xlarge)

bhalevy commented:
@scylladb/scylla-maint please consider merging.
CI failure is unrelated to this change.

denesb commented:
We normally require green CI, but considering that the CI on this takes 11hours to run, only the rolling test failed (due to missing file streaming) and that this is a CI stability fix, I will make an exception and queue this.
